.github/workflows/canister-tests.yml

```diff
@@ -221,9 +221,7 @@ jobs:
       # Make sure that one failing test does not cancel all other matrix jobs
       fail-fast: false
     env:
-      DFX_VERSION: 0.8.3
-      REPLICA_URL: https://ic0.app
-      II_ORIGIN: https://identity.ic0.app
+      DFX_VERSION: 0.11.1
```

Comment:
do we actually need the newer version to run this? should we do this separately and update `using-dev-build` as well?

Reply:
If I leave out the `null` deployment argument as suggested below then we need it, right?
I would suggest to leave the version here and I will create a PR shortly also updating `using-dev-build` to the same version.

Reply:
sounds good. At the same time if you feel like creating an action for setting up dfx... that'd be amazing 😬 In particular it would fix two issues:
* `dfx` could be cached, which would prevent somewhat regular transient issues when the sdk download URL returns 503 (and might even save a few seconds, although that's secondary)
* We'd have a single place where the version is defined